pyccel/ast/operators.py

```diff
@@ -488,24 +511,46 @@ class PyccelAdd(PyccelArithmeticOperator):
     arg2: PyccelAstNode
         The second argument passed to the operator
     """
-    __slots__ = ()
+    __slots__ = ('_dtype','_precision',)
```

Comment:
Classes inherit slots from the classes that they inherit from so `PyccelAdd` already has a `_dtype` and `_precision` as they are inherited from `PyccelArithmeticOperator` which inherits them from `PyccelBinaryOperator`